### ISSUE TITLE
dp.change not firing after a dp.error (2nd and subsequent times) when you correct the invalid date using keyboard

### ISSUE THREAD

**Issue by dhanlon** [Eonasdan/bootstrap-datetimepicker#1628]
I am using version  4.17.42.

Follow this process to see the bug.

Using the picker with keepInvalid = true, enter a valid date, say '2016-05-26'.

Now do the following 2 things. 
1. Using the keyboard insert an 'a' character somewhere in the date text to make it invalid. A dp.error event is raised. Anything can be used here.
2. Now use the keyboard to remove this invalid character and make date valid again. A dp.change event is raised.

But now repeat steps 1. and 2., the 'dp.error' event will happen in step 1, but no 'dp.change' event will happen in step 2. If you were to correct the date in step 2 using the picker, the 'dp.change' event would be raised.

A fiddle (my first one... hope I have this set up right :) to show this is here:  http://jsfiddle.net/DavidHanlon/jd80j4hc/

I am just logging the dp.change and dp.error events to the Chrome console.

I am using the picker with knockout and knockout validation. Following the steps above leaves my model value out of sync with the text box value, and the knockout validation error does not clear after making the correction because the binding handler is not aware of the updated value in the text box.

**Comment by podger** [comment]
Same issue here! 
Did you managed to work around it someway?

**Comment by dhanlon** [comment]
We have a workaround, but it isn't very satisfying.

We set keepInvalid=false so that the picker will automatically correct an invalid typed in date. We updated the knockout binding handler to detect when this happens and set an observable subproperty when it does.  Using this subproperty we can display a warning message to the end user to alert them that the value they had just entered was invalid and adjusted. Pretty lame but the best we can do.
